### ISSUE TITLE
Support horizontal scrollbar?

### ISSUE THREAD

**Issue by thovo** [malte-wessel/react-custom-scrollbars#78]
Thank you a lot for the component! It worked pretty nice. 

**Situation**: I am using `overflow-x: auto` to display a horizontal scrollbar for **Tabs** component of `react-toolbox`. I want to use `react-custom-scrollbar` for the `nav` part of **Tabs** component.

I want to know there is any support for horizontal scrolling now and if it has, how to implement it?

I tried to look at the documentation and around issues discussion but I didn't find any example using it. I am extremely grateful if you can show me how to do it.

**Comment by malte-wessel** [comment]
You'll need to make the Scrollbars' child components wider than the Scrollbars component itself. This can be done with a fixed width on the child components.

**Comment by thovo** [comment]
The problem is usually we don't know the available width of child component as the data would be very random.

**Comment by malte-wessel** [comment]
Maybe you could force the tabs to overflow with the help of css. If you got `inline-block` elements you could set `white-space: nowrap;` on the tabs container element, to ensure that the container grows horizontally.

**Comment by thovo** [comment]
Yes, I am doing it now, I force the `nav` inside `Tabs` have `overflow-x: auto` so it will add a horizontal scrollbar now if it is too long. The problem is wrap the whole `Tabs` component inside `Scrollbars` didn't render the nice scroll but always the default scrollbar of browser which is a little ugly.

An example is below:
![tabs](https://cloud.githubusercontent.com/assets/1543066/18266824/bba81388-741c-11e6-8fba-35730462640c.png)

**Comment by janejanejane** [comment]
I don't understand how this is used to show the horizontal scrollbar.

**Comment by thovo** [comment]
@janejanejane : it didn't show yet and currently I don't know how, to enable the horizontal scrollbar you just need to add style `overflow-x: auto` to the element you want.

**Comment by WonderStudio** [comment]
@thovo 

React file :
`<CustomScrollBars 
autoHide={false}
style={{ width: "100%", height: "48px" }}
renderTrackHorizontal={props => <div {...props} className="track-horizontal"/>}
renderThumbHorizontal={props => <div {...props} className="thumb-horizontal"/>}>
</WFScrollBars>`

CSS file :
`.track-horizontal {
    min-width: 100%;
    z-index: 150;
    bottom: 0;
}
.thumb-horizontal {
    cursor: pointer;
    background: red;
    min-width: 100px;
}`

**Comment by ignu** [comment]
I tried that in a codesandbox and it looks like it's being covered up by native scrollbars?

https://codesandbox.io/s/nZm4PKxPp

cc: @WonderStudio

**Comment by CaptainN** [comment]
You can use something like `react-measure` to get the width of the container, and use that width to set the width of the scroll bar.

```JSX
<Measure bounds onResize={this.handleResize}>
    {({measureRef}) => <div ref={measureRef} className="chooser-objects"
      onWheel={this.handleWheel}>
      <Scrollbars universal style={{ width: dimensions.width, height: dimensions.height }}>
```

The real challenge is getting it to convert vertical scroll motion to horizontal, for mouse wheels.

**Comment by MuYunyun** [comment]
Can it support to show the vertical and the horizontal scrollbar at the same time?

**Comment by HUGY1** [comment]
i have done it.  using css  become horizontal scrollbar. and use onWheel eventlistener to scroll by mousewheel. looks good.
